### PR TITLE
Improves bottom sheet accessibility with larger font sizes

### PR DESCRIPTION
**Pull request by emilylaguna**
**WPiOS Related PR:** https://github.com/wordpress-mobile/WordPress-iOS/pull/16501

This PR does the following:
- Improves the intrinsic content calculations by calling `layoutIfNeeded` on the presented view before asking for the `preferredContentSize`. 
- On iPad if the user has a "larger font" enabled the bottom sheet will be presented as a `formSheet` to prevent any clipping issues in the popover
- This also removes the padding in the `computePaddedPreferredContentSize` since it is adding too much extra height to the bottom of the views. The `additionalContentTopMargin` is left in to maintain compatibility with Woo iOS

### WPiOS Testing Steps:

Follow the steps in the related PR:
- https://github.com/wordpress-mobile/WordPress-iOS/pull/16501

### WooCommerce iOS Testing Steps
1. Edit the PodFile to point to this branch
2. Launch the app on iPhone
3. Tap the Products tab
4. Tap the 'Sort By' option
5. Verify the bottom sheet opens and is not cut off
6. Tap the 'Add Product' button or `+` button  
5. Verify the bottom sheet opens and is not cut off
6. Increase the font size using the steps in the related PR
7. Verify the bottom sheets open and are not cut off
8. Increase the font size to the larger text size
9. Verify the bottom sheet opens and is not cut off
10. Repeat the steps on iPad
11. When enabling the larger text size
12. Verify the view is opened as a sheet in the middle of the screen

### CODE REVIEW

File: WordPressUI/BottomSheet/BottomSheetViewController.swift [hunk 1]
```diff
@@ -10,8 +10,8 @@ public class BottomSheetViewController: UIViewController {
         /// The height of the space above the bottom sheet content, including the grip view and space around it.
         ///
         public static let additionalContentTopMargin: CGFloat = BottomSheetViewController.Constants.gripHeight
-            + BottomSheetViewController.Constants.Header.spacing
-            + BottomSheetViewController.Constants.Stack.insets.top
+        + BottomSheetViewController.Constants.Header.spacing
+                    + BottomSheetViewController.Constants.Stack.insets.top
```
Comment (aerych) [inline WordPressUI/BottomSheet/BottomSheetViewController.swift]: Was this change intentional? Seems like things would look misaligned now?

Comment (emilylaguna) [reply WordPressUI/BottomSheet/BottomSheetViewController.swift]: Hm, definitely not intentional. Fixed: 7696cdbf434b5da54f835c2a85d25a2e2774352d

**Review by frosty — APPROVED** [review]
Thank you for working on this!
